app/src/main/java/io/github/drp08/studypal/presentation/screens/ProfileScreen.kt

```diff
@@ -4,17 +4,423 @@ import androidx.compose.material3.Text
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import co.yml.charts.axis.AxisData
+import co.yml.charts.axis.DataCategoryOptions
+import co.yml.charts.common.model.Point
+import co.yml.charts.ui.barchart.BarChart
+import co.yml.charts.ui.barchart.models.BarChartData
+import co.yml.charts.ui.barchart.models.BarChartType
+import co.yml.charts.ui.barchart.models.BarData
+import co.yml.charts.ui.barchart.models.BarStyle
+import kotlin.random.Random
 
 data object ProfileScreen : Screen {
     @Composable
     override fun Content() {
         val viewModel = hiltViewModel<ProfileViewModel>()
 
         val user = ActiveUser.current
+        Column(
+            modifier = Modifier
+                .verticalScroll(state = rememberScrollState())
+                .fillMaxWidth()
+                .padding(start = 10.dp, top = 10.dp, end = 10.dp, bottom = 20.dp),
+            verticalArrangement = Arrangement.Top,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Row(
+                modifier = Modifier
+                    .padding(start = 20.dp, top = 10.dp, end = 10.dp, bottom = 5.dp)
+                    .fillMaxWidth(),
+                horizontalArrangement = Arrangement.Start,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Icon(Icons.Default.Face , contentDescription = "Student icon",
+                    modifier = Modifier.size(128.dp))
+                Spacer(modifier = Modifier.width(30.dp))
+                Column (
+                    modifier = Modifier
+                        .fillMaxHeight(),
+                    verticalArrangement = Arrangement.Top,
+                    horizontalAlignment = Alignment.Start
+                ) {
+                    Text("Name: ${user.name}", fontSize = 24.sp, color = Color.DarkGray)
+                    Spacer(modifier = Modifier.height(12.dp))
+                    Text("Total XP: ")
+                }
+            }
+            Row (
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 20.dp, end = 20.dp, bottom = 10.dp),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                OutlinedButton(
+                    onClick = { ProfileScreen}
+                ) {
+                    Text("Add Friends")
+                }
+                Spacer(modifier = Modifier.width(45.dp))
+                OutlinedButton(
+                    onClick = { ProfileScreen }
+                ) {
+                    Text("Your Friends")
+                }
+            }
+            var numTopicsExpanded by remember { mutableIntStateOf(0) }
+            var numberSubjects by remember { mutableIntStateOf(0) }
+            val subjectList = arrayOf("Statistics") //Todo: Change this to the actual list of subjects
```

Comment:
The main code has code to get an actual list of subjects.. Can you fetch that?

Reply:
The reason I'm pushing is cause Bella's offered to help me do that, but needs to pulled and merged with main first - so will be done after - but yes does need to be done

Reply:
Cool okay!

Reply:
It's not breaking anything dw